components/MainLayout.js

```diff
@@ -1,3 +1,9 @@
+/*
+eslint
```

Comment:
Since these rule overrides only apply to the smooth scrolling container (not the whole MainLayout), I think it is better to move this rule overrides and associated logic to its own file.

Please extract `<div onClick={handleLinkClick}>` into another React component  `<SmoothScrollContainer>` in its own file, so that it is clear that we need to add these rules for these purposes.